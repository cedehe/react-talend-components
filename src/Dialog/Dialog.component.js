import React, { PropTypes } from 'react';
import { Modal } from 'react-bootstrap';
import Action from '../Actions/Action';

/**
 * @param {object} props react props
 * @example
<Dialog name="Hello world"></Dialog>
 */
function Dialog(props) {
	return (
		<Modal
			bsSize={props.size || null}
			show={props.show}
		>
			{props.header ? (
				<Modal.Header closeButton>
					<Modal.Title>{props.header}</Modal.Title>
				</Modal.Header>
			) : null}
			<Modal.Body>
				{props.children}
			</Modal.Body>
			{props.action ? (
				<Modal.Footer>
					<Action {...props.action} />
				</Modal.Footer>
			) : null}
		</Modal>
	);
}

Dialog.propTypes = {
	header: PropTypes.string,
	size: PropTypes.oneOf(['small', 'large']),
	children: PropTypes.element,
	show: PropTypes.bool,
	action: PropTypes.shape(Action.propTypes),
};

export default Dialog;
