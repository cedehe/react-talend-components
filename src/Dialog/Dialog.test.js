import React, { PropTypes } from 'react';

jest.mock('react-dom');
jest.mock('react-bootstrap', () => {
	function getFakeComponent(name) {
		return ({ children, ...rest}) => (<div className={name} {...rest}>{children}</div>);
	}
	const Modal = getFakeComponent('Modal');
	Modal.Header = getFakeComponent('Header');
	Modal.Title = getFakeComponent('Title');
	Modal.Body = getFakeComponent('Body');
	Modal.Footer = getFakeComponent('Footer');
	const OverlayTrigger = getFakeComponent('OverlayTrigger');
	OverlayTrigger.propTypes = {};
	const Tooltip = getFakeComponent('Tooltip');
	function Button({ children, ...rest }) {
		return (<button {...rest}>{children}</button>);
	}
	return {
		Button,
		Modal,
		OverlayTrigger,
		Tooltip,
	};
});

import renderer from 'react-test-renderer';

import Dialog from './Dialog.component';

const defaultProps = {
	show: true,
};
const headerProps = {
	header: 'Hello world',
	show: true,
};
const actionProps = {
	show: true,
	header: 'Hello world',
	action: {
		label: 'OK',
		onClick: jest.fn(),
	},
};
const smallProps = {
	show: true,
	header: 'Hello world',
	size: 'small',
	action: {
		label: 'OK',
		onClick: jest.fn(),
	},
};
const largeProps = {
	show: true,
	header: 'Hello world',
	size: 'large',
	action: {
		label: 'OK',
		onClick: jest.fn(),
	},
};
const children = (<div>BODY</div>);

describe('Dialog', () => {
	it('should render', () => {
		const wrapper = renderer.create(
			<Dialog {...defaultProps}>{children}</Dialog>
		).toJSON();
		expect(wrapper).toMatchSnapshot();
	});
	it('should render header', () => {
		const wrapper = renderer.create(
			<Dialog {...headerProps}>{children}</Dialog>
		).toJSON();
		expect(wrapper).toMatchSnapshot();
	});
	it('should render action', () => {
		const wrapper = renderer.create(
			<Dialog {...actionProps}>{children}</Dialog>
		).toJSON();
		expect(wrapper).toMatchSnapshot();
	});
	it('should render small', () => {
		const wrapper = renderer.create(
			<Dialog {...smallProps}>{children}</Dialog>
		).toJSON();
		expect(wrapper).toMatchSnapshot();
	});
	it('should render large', () => {
		const wrapper = renderer.create(
			<Dialog {...largeProps}>{children}</Dialog>
		).toJSON();
		expect(wrapper).toMatchSnapshot();
	});
});
