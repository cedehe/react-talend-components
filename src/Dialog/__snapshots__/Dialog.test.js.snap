exports[`Dialog should render 1`] = `null`;

exports[`Dialog should render action 1`] = `null`;

exports[`Dialog should render header 1`] = `null`;

exports[`Dialog should render large 1`] = `null`;

exports[`Dialog should render small 1`] = `null`;
